nengo_gui/components/value.py

```diff
@@ -59,6 +59,9 @@ def update_client(self, client):
         # while there is data that should be sent to the client
         while len(self.data) > 0:
             item = self.data.popleft()
+            while len(self.data) > 0:
```

Comment:
You could also change the first `while` to an `if`. I think, that would make it already better. I'm not sure what the datatype of `item` is (just looking at the changed code excerpt), but maybe something like
```python
item = ''
while len(self.data) > 0:
    item += self.data.popleft()
```
would work?